lib/manager/setup-cfg/extract.ts

```diff
@@ -25,64 +27,88 @@ function getDepType(section: string, record: string): null | string {
       return 'test';
     }
   }
-  return 'extra';
+  if (section === 'options.extras_require') {
+    return 'extra';
+  }
+  return null;
 }
 
 function parseDep(
   line: string,
   section: string,
   record: string
 ): PackageDependency | null {
-  const [, depName, , currentValue] =
-    regEx(/\s+([-_a-zA-Z0-9]*)(\[.*\])?\s*(.*)/).exec(line) || [];
-  if (
-    section &&
-    record &&
-    depName &&
-    currentValue &&
-    pep440.isValid(currentValue)
-  ) {
-    const dep: PackageDependency = {
-      datasource: PypiDatasource.id,
-      depName,
-      currentValue,
-    };
-    const depType = getDepType(section, record);
-    if (depType) {
-      dep.depType = depType;
-    }
-    return dep;
+  const packagePattern = '[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9]';
+  const extrasPattern = '(?:\\s*\\[[^\\]]+\\])?';
+
+  const rangePattern: string = RANGE_PATTERN;
+  const specifierPartPattern = `\\s*${rangePattern.replace(
+    regEx(/\?<\w+>/g),
+    '?:'
+  )}`;
+  const specifierPattern = `${specifierPartPattern}(?:\\s*,${specifierPartPattern})*`;
+  const dependencyPattern = `(${packagePattern})(${extrasPattern})(${specifierPattern})`;
+
+  const pkgRegex = regEx(`^(${packagePattern})$`);
+  const pkgValRegex = regEx(`^${dependencyPattern}$`);
+
+  const depType = getDepType(section, record);
+  if (!depType) {
+    return null;
   }
-  return null;
+
+  const [lineNoEnvMarkers] = line.split(';').map((part) => part.trim());
+  const packageMatches =
+    pkgValRegex.exec(lineNoEnvMarkers) || pkgRegex.exec(lineNoEnvMarkers);
+
+  if (!packageMatches) {
+    return null;
+  }
+
+  const [, depName, , currVal] = packageMatches;
+  const currentValue = currVal?.trim();
+
+  const dep: PackageDependency = {
+    depName,
+    currentValue,
+    datasource: PypiDatasource.id,
+    depType: depType,
+  };
+
+  if (currentValue?.startsWith('==')) {
+    dep.currentVersion = currentValue.replace(/^==\s*/, '');
+  }
+
+  return dep;
 }
 
 export function extractPackageFile(
   content: string
 ): Result<PackageFile | null> {
+  logger.trace('setup-cfg.extractPackageFile()');
+
   let sectionName = null;
   let sectionRecord = null;
 
   const deps: PackageDependency[] = [];
-  content
-    .split(newlineRegex)
-    .map((line) => line.replace(regEx(/[;#].*$/), '').trimRight())
-    .forEach((rawLine) => {
-      let line = rawLine;
-      const newSectionName = getSectionName(line);
-      const newSectionRecord = getSectionRecord(line);
-      if (newSectionName) {
-        sectionName = newSectionName;
-      } else {
-        if (newSectionRecord) {
-          sectionRecord = newSectionRecord;
-          line = rawLine.replace(regEx(/^[^=]*=\s*/), '\t');
-        }
-        const dep = parseDep(line, sectionName, sectionRecord);
-        if (dep) {
-          deps.push(dep);
-        }
-      }
-    });
-
-  return deps.length > 0 ? { deps } : null;
+  content.split(newlineRegex).map((rawLine) => {
```

Comment:
Why you changed this? Comment line should still skipped, also use forEach instead of map when you don't use the result.

Reply:
I will add more test to check this behavior.

Reply:
fixed